numpyro/infer/mcmc.py

```diff
@@ -507,23 +517,10 @@ def init(self, rng_key, num_warmup, init_params=None, model_args=(), model_kwarg
         # vectorized
         else:
             rng_key, rng_key_init_model = np.swapaxes(vmap(random.split)(rng_key), 0, 1)
-            # we need only a single key for initializing PE / constraints fn
-            rng_key_init_model = rng_key_init_model[0]
-        if not self._init_fn:
-            self._init_state(rng_key_init_model, model_args, model_kwargs)
+        init_params = self._init_state(rng_key_init_model, model_args, model_kwargs, init_params)
         if self._potential_fn and init_params is None:
             raise ValueError('Valid value of `init_params` must be provided with'
                              ' `potential_fn`.')
-        # Find valid initial params
-        if self._model and not init_params:
-            init_params, is_valid = find_valid_initial_params(rng_key, self._model,
-                                                              init_strategy=self._init_strategy,
-                                                              model_args=model_args,
-                                                              model_kwargs=model_kwargs)
-            if not_jax_tracer(is_valid):
-                if device_get(~np.all(is_valid)):
-                    raise RuntimeError("Cannot find valid initial parameters. "
-                                       "Please check your model again.")
```

Comment:
I have moved those logics to `init_state` method. The only behavior change here is: `init_params` arg takes no effect when `self._model is not None`. If we want to preserve the old behavior (i.e. using `init_params` for `model` and skip `find_valid_inititial_params`, we can simply add an args `init_params` to `initialize_model` class).